### PR TITLE
Allow duplication of events

### PR DESCRIPTION
**Pull request by david-venhoff**
### Short description
This pr allows the user to duplicate events on the event list screen.

### Proposed changes
Adds a button on event_list_row to duplicate the selected event

Fixes: #593

### CODE REVIEW

File: src/cms/models/events/event.py [hunk 1]
```diff
@@ -224,6 +225,69 @@ def best_translation(self):
...
+        self.save()
+
+        copy_translation = _("copy")
+        # Create new translations for this event
+        for translation in translations:
+            translation.pk = None
+            translation.event = self
+            translation.status = status.DRAFT
+            translation.title = f"{translation.title} ({copy_translation})"
+            translation.slug = generate_unique_slug(
```
Comment (david-venhoff) [inline src/cms/models/events/event.py:272]: I am not quite certain to have used this function correctly. Using the same `slug` as `translation.slug`, I get duplicate slugs under some circumstances.

File: src/cms/models/events/event.py [hunk 2]
```diff
@@ -224,6 +225,69 @@ def best_translation(self):
         """
         return self.backend_translation or self.default_translation
 
+    def duplicate(self, user, region, language):
```
Comment (timoludwig) [inline src/cms/models/events/event.py:228]:
```suggestion
    def duplicate(self, user):
```

File: src/cms/models/events/event.py [hunk 3]
```diff
@@ -224,6 +225,69 @@ def best_translation(self):
...
+
+        :param user: The user who initiated this copy
+        :type user: ~django.contrib.auth.models.User
+
+        :param region: The current region, used to generated unique slugs
+        :type region: ~cms.models.regions.region.Region
+
+        :param language: The current language, used to generate unique slugs
+        :type language: ~cms.models.languages.language.Language
+
```
Comment (timoludwig) [inline src/cms/models/events/event.py:241]:
```suggestion
```

File: src/cms/views/events/event_actions.py [hunk 4]
```diff
@@ -64,6 +64,46 @@ def archive(request, event_id, region_slug, language_slug):
...
+    :rtype: ~django.template.response.TemplateResponse
+    """
+    region = Region.get_current_region(request)
+    event = get_object_or_404(region.events, id=event_id)
+    language = get_object_or_404(region.languages, slug=language_slug)
+
+    if not request.user.has_perm("cms.edit_events"):
+        raise PermissionDenied
+
+    event.duplicate(request.user, region, language)
```
Comment (timoludwig) [inline src/cms/views/events/event_actions.py:97]:
```suggestion
    event.duplicate(request.user)
```

**Review by timoludwig — COMMENTED** [review]
Thank you! :+1:
In general this works, see my comments below for a few improvements.

**Review by timoludwig — APPROVED** [review]
Thank you! :rocket:

**Comment by timoludwig** [comment]
If you want to update your branch with current changes from develop, please use rebasing instead of merging in the future (see [our documentation](https://integreat.github.io/integreat-cms/git-flow.html#update-feature-branch)).

**Pull request merged**